packages/framework/src/Component/Router/FriendlyUrl/FriendlyUrlFactory.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Component\Router\FriendlyUrl;
+
+class FriendlyUrlFactory implements FriendlyUrlFactoryInterface
+{
+
+    /**
+     * @param string $routeName
+     * @param int $entityId
+     * @param int $domainId
+     * @param string $slug
+     * @return \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrl
+     */
+    public function create(
+        string $routeName,
+        int $entityId,
+        int $domainId,
+        string $slug
+    ): FriendlyUrl {
+        return new FriendlyUrl($routeName, $entityId, $domainId, $slug);
```

Comment:
I noticed that there are few entities that do not have corresponding DataObject. Could you please create a new user-story with a list of such entities? I think we should unify the data-object approach or at least discuss it sometime later

Reply:
This problem is trickier, for example `AdministratorActivity` expects `Administrator $administrator, string $ipAddress`.
These information could be eventually wrapped in data object, but should it be?

Reply:
This problem is trickier, for example `AdministratorActivity` expects `__construct(Administrator $administrator, string $ipAddress)`.
This also seem to be problem of all `*Domain` objects...
These information could be eventually wrapped in data object, but should it be?

Reply:
As we discussed earlier, I'll create a US for that, or maybe I'll just add a note into some existing one :wink: